src/notation/view/notationpaintview.cpp

```diff
@@ -388,20 +388,29 @@ std::pair<int, int> NotationPaintView::constraintCanvas(int dx, int dy) const
     QRectF viewport = this->viewport();
 
     QPoint canvasCenter = this->canvasCenter();
-    if (contentRect.width() <= viewport.width()) {
+
+    bool isScrollLimited = configuration()->isLimitCanvasScrollArea();
+
+    int offsetX = viewport.width() - 100;
+    int offsetY = viewport.height() - 100;
+    if (isScrollLimited) {
+        offsetX = offsetY = 100;
```

Comment:
I'm afraid this fix won't work on hidpi screens. Anyway, it is better don't use any magic numbers here. You should use adaptive values dependent on the current screen width/height/scaling

Reply:
Right. I'll change the 100 to a relative value. I don't understand why it won't work on hidpi screens though. Could you please explain why?

Reply:
This is just my assumption, I don't have any proofs :) But magic numbers always have such kinds of problems (especially in the score view where everything depends on the current screen configuration). So better always use relative values